.travis.yml

```diff
@@ -13,25 +13,55 @@ notifications:
-    - ROS_DISTRO=kinetic ROS_REPO=ros-shadow-fixed UPSTREAM_WORKSPACE=travis.rosinstall BEFORE_SCRIPT="echo 'testing'"
-    - ROS_DISTRO=kinetic ROS_REPO=ros-shadow-fixed
-      TEST_BLACKLIST=moveit,moveit_ros,moveit_runtime,moveit_ros_perception
-      UPSTREAM_WORKSPACE="https://raw.githubusercontent.com/ros-planning/moveit/kinetic-devel/moveit.rosinstall,
-                          https://raw.githubusercontent.com/ros-planning/moveit/kinetic-devel/moveit.rosinstall"
+    # default success case, successful BEFORE_SCRIPT
+    - BEFORE_SCRIPT="echo first && false; echo second > /dev/null; echo Testing on $ROS_DISTRO"
+    - UPSTREAM_WORKSPACE=travis.rosinstall
+    - CI_SOURCE_PATH=/root/moveit_ci/test_pkgs/valid     TEST=clang-format
+
+    # failures: result=2 indicates terminate of Travis script
+    - EXPECTED_RESULT=2  BEFORE_SCRIPT="echo \"Failing BEFORE_SCRIPT\"; return 1"
+    - EXPECTED_RESULT=2  UPSTREAM_WORKSPACE=missing.rosinstall  # missing UPSTREAM_WORKSPACE
+    - EXPECTED_RESULT=2  TEST=invalid # invalid test
+    - EXPECTED_RESULT=0  CI_SOURCE_PATH=/root/moveit_ci/test_pkgs/warnings      WARNINGS_OK=true
+    - EXPECTED_RESULT=2  CI_SOURCE_PATH=/root/moveit_ci/test_pkgs/warnings      WARNINGS_OK=false
+    - EXPECTED_RESULT=0  CI_SOURCE_PATH=/root/moveit_ci/test_pkgs/valid         WARNINGS_OK=false
+    - EXPECTED_RESULT=2  CI_SOURCE_PATH=/root/moveit_ci/test_pkgs/catkin_lint   TEST=catkin_lint
+    - EXPECTED_RESULT=2  CI_SOURCE_PATH=/root/moveit_ci/test_pkgs/clang_format  TEST=clang-format
+    - EXPECTED_RESULT=2  CI_SOURCE_PATH=/root/moveit_ci/test_pkgs/clang_tidy    TEST="clang-tidy-check clang-tidy-fix"
+    - EXPECTED_RESULT=2  CI_SOURCE_PATH=/root/moveit_ci/test_pkgs/clang_format  TEST=clang-format                        ROS_DISTRO=kinetic
+    - EXPECTED_RESULT=2  CI_SOURCE_PATH=/root/moveit_ci/test_pkgs/clang_tidy    TEST="clang-tidy-check clang-tidy-fix"   ROS_DISTRO=kinetic
+
+    # Build MoveIt! repository
+    - CI_SOURCE_PATH=/root/ws_moveit/src/moveit   TEST_BLACKLIST=moveit_ros_perception
+      UPSTREAM_WORKSPACE="https://raw.githubusercontent.com/ros-planning/moveit/$ROS_DISTRO-devel/moveit.rosinstall,
+                          https://raw.githubusercontent.com/ros-planning/moveit/$ROS_DISTRO-devel/moveit.rosinstall"
                           # Duplicated rosinstall file for testing purposes
 
+    # Build geometric_shapes and apply clang-tidy-fix
+    - CI_SOURCE_PATH=/root/ws_moveit/src/geometric_shapes  TEST=clang-tidy-fix
+      UPSTREAM_WORKSPACE=https://github.com/ros-planning/geometric_shapes
+
+matrix:
+  include: # Build MoveIt! repository with clang + clang-tidy
+    - env: CI_SOURCE_PATH=/root/ws_moveit/src/moveit    TEST_BLACKLIST=moveit_ros_perception    TEST=clang-tidy-check
+           UPSTREAM_WORKSPACE=https://raw.githubusercontent.com/ros-planning/moveit/$ROS_DISTRO-devel/moveit.rosinstall
+      compiler: clang
+
+  allow_failures:
```

Comment:
add comment how "allow_failures" is different than the EXPECTED_RESULT stuff above